consensus/pbft/viewchange.go

```diff
@@ -435,23 +435,23 @@ func (instance *pbftCore) processNewView() events.Event {
 				continue
 			}
 
-			if _, ok := instance.reqStore[d]; !ok {
-				logger.Warningf("Replica %d missing assigned, non-checkpointed request %s",
+			if _, ok := instance.reqBatchStore[d]; !ok {
+				logger.Warningf("Replica %d missing assigned, non-checkpointed request batch %s",
 					instance.id, d)
-				if _, ok := instance.missingReqs[d]; !ok {
-					logger.Warningf("Replica %v requesting to fetch %s",
+				if _, ok := instance.missingReqBatches[d]; !ok {
+					logger.Warningf("Replica %v requesting to fetch batch %s",
 						instance.id, d)
-					newRequestMissing = true
-					instance.missingReqs[d] = true
+					newReqBatchMissing = true
+					instance.missingReqBatches[d] = true
 				}
 			}
 		}
 	}
 
-	if len(instance.missingReqs) == 0 {
+	if len(instance.missingReqBatches) == 0 {
 		return instance.processNewView2(nv)
-	} else if newRequestMissing {
-		instance.fetchRequests()
+	} else if newReqBatchMissing {
+		instance.fetchRequestBatches()
```

Comment:
I thought this was standard protocol behavior.

On 07/21/2016 05:08 PM, Jason Yellick wrote:
>
> In consensus/pbft/viewchange.go
> <https://github.com/hyperledger/fabric/pull/2234#discussion_r71724052>:
>
> >  		return instance.processNewView2(nv)
> > -	} else if newRequestMissing {
> > -		instance.fetchRequests()
> > +	} else if newReqBatchMissing {
> > +		instance.fetchRequestBatches()
>
> Well, we are already deviating by fetching missing requests, which
> will not necessarily succeed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hyperledger/fabric/pull/2234/files/990bf439bdf841d2390c64e9b756de130d7e2dd1#r71724052>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAK3O1v_D2JkLtW-sFWjAE7j9ckUm72Gks5qX4uDgaJpZM4JOH5S>.
>